scout/adapter/mongo/hgnc.py

```diff
@@ -317,13 +316,24 @@ def hgncsymbol_to_gene(self, build="37", genes=None):
         LOG.info("All genes fetched")
         return hgnc_dict
 
-    def gene_by_alias(self, symbol, build="37"):
-        """Return an iterable with hgnc_genes.
+    def gene_by_symbol_or_aliases(self, symbol, build="37"):
+        """Return an iterable with only one gene when gene with a given symbol if found
+           or a cursor with genes where the provided symbol is among the aliases.
+        Args:
+            symbol(str)
+            build(str)
+
+        Returns:
+            res(list or pymongo.Cursor(dict)): return a list with one gene or a cursor with several gene dictionaries
+        """
+        res = self.hgnc_collection.find_one({"hgnc_symbol": symbol, "build": str(build)})
+        if res:
+            return [res]
 
-        If the gene symbol is listed as primary the iterable will only have
-        one result. If not the iterable will include all hgnc genes that have
-        the symbol as an alias.
+        return self.gene_by_alias(symbol, build="37")
 
+    def gene_by_alias(self, symbol, build="37"):
```

Comment:
Hm, isn't the convention that we always find_one with the gene_ but allow multiple with the geneS_?

Reply:
It should have been genes_by_alias before really.. 😁

Reply:
I think it would be better to return them all in this case, because you don't know which alias the right one is

Reply:
there is another function, down, that is named geneS_by_alias.. 😆

Reply:
We have to rename the other function in case

Reply:
But I agree, should be geneS